src/online.jl

```diff
@@ -1,40 +1,63 @@
 import Base: append!, push!
 
-function push!(di::LinearInterpolation{U, T}, u::eltype(U), t::eltype(T)) where {U, T}
-    push!(di.u, u)
-    push!(di.t, t)
-    di
+function push!(A::LinearInterpolation{U, T}, u::eltype(U), t::eltype(T)) where {U, T}
```

Comment:
Shouldn't this be allowed only if `safetycopy` is false?

Reply:
I don't think I fully understand the reasoning here. If `safetycopy` is true, then the `u, t` vectors within the interpolation object can safely be assumed to be only used within the context of this package. Surely then it is safe to add data to it? Only if `u, t` are also used in code outside the `DataInterpolations` context do you run the risk of breaking that code by modifying `u, t`.

Reply:
I think I understand your point. It comes down what should be the semantics of `push!` for interpolation objects. Currently it mutates the original array. Maybe we should handle both cases? If it is true, it would push to the copy, if it is false it would mutate the original array. For both we have to compute the parameters again. 
@ChrisRackauckas, your thoughts?